### PR TITLE
Attentive suggestions

### PR DESCRIPTION
**Pull request by galvana**
### Description Of Changes

Example of how to handle optional identities + misc cleanup suggestions

### CODE REVIEW

File: data/saas/config/attentive_config.yml [hunk 1]
```diff
@@ -23,15 +23,15 @@ saas_config:
     path: /v1/me
 
   endpoints:
-    - name: delete_request
+    - name: user
```
Comment (galvana) [inline data/saas/config/attentive_config.yml:26]: Same as the feedback on PowerReviews, we want the endpoint name to reflect the model type that this represents, in this case `user`.

File: data/saas/config/attentive_config.yml [hunk 2]
```diff
@@ -23,15 +23,15 @@ saas_config:
...
       requests:
         delete:
           method: POST
           path: /v1/privacy/delete-request
           body: |
             {
-             "subjectEmail": "<email>",
-             "subjectPhone": "<phone_number>",
+             "subjectEmail": "<email?>",
+             "subjectPhone": "<phone_number?>",
```
Comment (galvana) [inline data/saas/config/attentive_config.yml:34]: We have this `?` notation in our placeholders to indicate that this value might be `null` but it's ok! I added this here so there aren't any errors if either `email` or `phone_number` are missing

File: data/saas/dataset/attentive_dataset.yml [hunk 3]
```diff
@@ -2,34 +2,4 @@ dataset:
...
-              data_type: string
-          - name: processingStartDateTime
-            data_categories: [system.operations]
-            fidesops_meta:
-              data_type: string
-          - name: processingEndDateTime
-            data_categories: [system.operations]
-            fidesops_meta:
-              data_type: string
+    collections: []
```
Comment (galvana) [inline data/saas/dataset/attentive_dataset.yml:5]: We can just set `collections` to `[]` if we don't have a `read` request

File: tests/ops/integration_tests/test_attentive_task.py [hunk 4]
```diff
@@ -25,8 +24,25 @@ async def test_non_strict_erasure_request(
             erasure_policy=erasure_policy_string_rewrite,
             identities={
                 "email": attentive_erasure_identity_email,
+            },
+        )
+        assert erasure_results == {"attentive_instance:user": 1}
+
+    async def test_non_strict_erasure_request_with_phone_number(
```
Comment (galvana) [inline tests/ops/integration_tests/test_attentive_task.py:31]: I added two separate tests to verify that the delete will work with either email or phone number

**Review by Vagoasdf — APPROVED** [review]

**Pull request merged**